src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.EasyRequest.cs

```diff
@@ -285,14 +289,15 @@ private void SetCredentialsOptions()
                 VerboseTrace("Set credentials options");
             }
 
-            private void SetCookieOption()
+            internal void SetCookieOption(Uri uri)
             {
                 if (!_handler._useCookie)
                 {
+                    SetCurlOption(CURLoption.CURLOPT_COOKIE, IntPtr.Zero);
```

Comment:
You were right. This portion was not necessary. The cookie reset was happening in the method that was taking care of the ```Location``` header.